tests/model_forms/tests.py

```diff
@@ -1764,6 +1764,30 @@ def test_model_multiple_choice_field(self):
         with self.assertRaises(ValidationError):
             f.clean([c6.id])
 
+    def test_model_multiple_choice_field_validate_choices_called_properly(self):
+        class CustomModelMultipleChoiceField(forms.ModelMultipleChoiceField, TestCase):
+            def validate_choices(self, queryset, field_name, selected_choices):
+                self.assertIsInstance(queryset, models.QuerySet)
+                self.assertQuerysetEqual(queryset.order_by('id'), [1], lambda a: a.id)
+                self.assertIsInstance(field_name, str)
+                self.assertEqual(field_name, 'pk')
+                self.assertIsInstance(selected_choices, frozenset)
+                self.assertEqual(selected_choices, frozenset([1]))
+                raise ValidationError('custom_error')
+
+        with self.assertNumQueries(1):
+            field = CustomModelMultipleChoiceField(Category.objects.all())
+            with self.assertRaises(ValidationError) as exc:
+                field.clean([self.c1.id])
+        self.assertEqual(exc.exception.message, 'custom_error')
+
+    def test_model_multiple_choice_field_validate_choices(self):
+        with self.assertNumQueries(1):
+            field = forms.ModelMultipleChoiceField(Category.objects.all())
+            with self.assertRaises(ValidationError) as exc:
```

Comment:
Use `assertRaisesMessage()`.

Reply:
I am not sure if we can use this, because I would like to assert to the exc.exception.params not the message

Reply:
True, `exc` -> `cm`.